samtranslator/model/sam_resources.py

```diff
@@ -2064,3 +2065,39 @@ def _add_connector_metadata(
             except KeyError:
                 original_metadata = {}
             resource.set_resource_attribute("Metadata", {**original_metadata, **metadata})
+
+
+class SamGraphQLApi(SamResourceMacro):
+    """SAM GraphQL API Macro (WIP)."""
+
+    resource_type = "AWS::Serverless::GraphQLApi"
+    property_types = {
+        "Name": PropertyType(False, IS_STR),
+        "Tags": PropertyType(False, IS_DICT),
+        "XrayEnabled": PropertyType(False, is_type(bool)),
+        "Auth": PropertyType(True, IS_DICT),
+    }
```

Comment:
If these are pass-throughs, should use `PassThroughProperty`:

https://github.com/aws/serverless-application-model/blob/96691c1b9e8d7133f72f8c9c95ecc1098e3d96f4/samtranslator/model/__init__.py#L57-L65

Also `PropertyType` is deprecated: https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines and:

https://github.com/aws/serverless-application-model/blob/c39c2807bbf327255de8abed8b8150b18c60f053/samtranslator/model/__init__.py#L16

Reply:
Thanks! Did not realize that PropertyType was deprecated. Also that reminds me that `Tags` is technically no longer a PassThrough, but i will change `XrayEnabled` to match that.